plugin/vcs/gitlab/gitlab.go

```diff
@@ -233,6 +249,46 @@ func (provider *Provider) TryLogin(ctx context.Context, oauthCtx common.OauthCon
 	return fetchUserInfo(ctx, oauthCtx, instanceURL, "user")
 }
 
+// FetchCommitByID fetch the commit data by id.
+func (provider *Provider) FetchCommitByID(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, commitID string) (*vcs.Commit, error) {
+	code, body, err := httpGet(
+		instanceURL,
+		fmt.Sprintf("projects/%s/repository/commits/%s", repositoryID, commitID),
+		&oauthCtx.AccessToken,
+		oauthContext{
+			ClientID:     oauthCtx.ClientID,
+			ClientSecret: oauthCtx.ClientSecret,
+			RefreshToken: oauthCtx.RefreshToken,
+		},
+		oauthCtx.Refresher,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch commit data on GitLab instance %s, err: %w", instanceURL, err)
+	}
+	if code >= 300 {
+		return nil, fmt.Errorf("failed to fetch commit data on GitLab instance %s, status code: %d",
+			instanceURL,
+			code,
+		)
+	}
+
+	var commit *Commit
```

Comment:
commit := &Commit{} ? Otherwise it's a nil pointer?

And Unmarshal should take a commit which is a pointer.

Reply:
Fixed.